packages/toolpad-components/src/Form.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Container as MUIContainer, ContainerProps, Button } from '@mui/material';
+import { createComponent } from '@mui/toolpad-core';
+import { Formik, useFormikContext } from 'formik';
+import { SX_PROP_HELPER_TEXT } from './constants';
+
+type FormValue = Record<string, string>;
+
+interface Props extends ContainerProps {
+  onSubmit: (values: any) => void;
+  submitLabel: string;
+  value: FormValue;
+  onChange: (value: FormValue) => void;
+}
+
+// Because: https://github.com/jaredpalmer/formik/issues/271
+function ChangeHandler({ onChange }: any) {
+  const { values } = useFormikContext();
+
+  React.useEffect(() => {
+    onChange(values);
+  }, [values, onChange]);
+
+  return null;
+}
+
+function Form({ children, onSubmit, onChange, submitLabel, sx, defaultValue, ...rest }: Props) {
+  return (
+    <MUIContainer disableGutters sx={sx} {...rest}>
+      <Formik
+        initialValues={defaultValue}
+        onSubmit={(values) => {
+          onSubmit(values);
+        }}
+      >
+        {(props) => (
+          <form onSubmit={props.handleSubmit}>
+            <MUIContainer />
+            <ChangeHandler onChange={onChange} />
+            {children}
+
+            <MUIContainer
+              disableGutters
+              sx={{
+                borderTop: 'solid 1px',
+                display: 'flex',
+                justifyContent: 'end',
+                width: '100%',
+                mt: 1,
+                pt: 1,
+              }}
+            >
+              <Button type="submit" variant="contained">
+                {submitLabel}
+              </Button>
+            </MUIContainer>
+          </form>
+        )}
+      </Formik>
+    </MUIContainer>
+  );
+}
+
+export default createComponent(Form, {
+  argTypes: {
+    children: {
+      visible: false,
+      typeDef: { type: 'element' },
+      control: { type: 'layoutSlot' },
+    },
+    value: {
```

Comment:
Maybe `values` instead of `value`? But that's just personal opinion so feel free to disagree , just my idea.

Reply:
I was thinking the same, but the idea was to keep pretty much the same API as any other form element has, i.e. `FilePicker` also provides either array of objects or object with multiple values, but it's main prop is still called `value` 🤔 I'm also not sure whether it should be `value` or `values`, both would be fine with me. Maybe @Janpot can tell the final word?

Reply:
I look at the `value`/`onChange` pattern as an abstract API for any controlled component. Simple ones, like `input`, and more complex ones, like our `MapEntriesEditor` that composes multiple controlled inputs.

The benefit lies in this being an abstract API. controlled inputs can easily be swapped by other ones, or composed into more complex ones, or wrapped by higher order APIs, or made dynamic and passed as a prop.

So from my side +1 on `value`/`onChange`